src/cookieSyncManager.ts

```diff
@@ -113,13 +126,18 @@ export default function CookieSyncManager(
 
             // Url for cookie sync pixel
             const fullUrl = createCookieSyncUrl(mpid, pixelUrl, redirectUrl)
-
-            self.performCookieSync(
-                fullUrl,
-                moduleId.toString(),
-                mpid,
-                cookieSyncDates
-            );
+            const moduleIdString = moduleId.toString();
+
+            if (isTcfApiAvailable()) {
```

Comment:
I see that this function calls with `performCookieSync` but seems to only define a callback and appends strings to the url. I see why you're doing it this way, but I wonder if it would be more readable (and possibly prevent side effects) if we break out the callback logic and have it simply return a string that then gets passed into `performCookieSync`. It might make things more obvious and less coupled.